src/lib/infiniteDiscovery/weaviate.ts

```diff
@@ -0,0 +1,251 @@
+import uuid from "uuid/v5"
+import gql from "../gql"
+import { random, sampleSize } from "lodash"
+
+interface UserResponse {
+  data: {
+    Get: {
+      InfiniteDiscoveryUsers: WeaviateUser[]
+    }
+  }
+}
+
+interface ArtworkResponse {
+  data: {
+    Get: {
+      InfiniteDiscoveryArtworks: { internalID: string }[]
+    }
+  }
+}
+
+interface WeaviateUser {
+  _additional: { id: string }
+  likedArtworks: { internalID: string }[]
+  seenArtworks: { internalID: string }[]
+}
+
+interface UserCreationBody {
+  class: string
+  id: string
+  properties: { internalID: string }
+}
+
+/**
+ * Extracts the first user from the Weaviate response
+ *
+ * @param  response of the image to read
+ * @returns the user object
+ */
+export function getUser(response: UserResponse): WeaviateUser | null {
+  const responseArray = response?.data?.Get
+
+  if (!responseArray) {
+    return null
+  }
+
+  return responseArray.InfiniteDiscoveryUsers[0]
+}
+
+/**
+ * Extracts the artwork internalIDs from the Weaviate response
+ *
+ * @param response from the Weaviate API containing InfiniteDiscoveryArtworks
+ * @returns an array of the artwork internalIDs
+ */
+export function GetArtworkIds(response: ArtworkResponse): string[] {
+  const responseArray = response?.data?.Get
+
+  if (!responseArray) {
+    return []
+  }
+
+  return responseArray.InfiniteDiscoveryArtworks.map(
+    (artwork) => artwork.internalID
+  )
+}
+
+/**
+ * Inserts a sample of curated artworks at random indexes into the target array
+ *
+ * @param target the array to insert the curated artworks into
+ * @param curatedWorks the array of curated artworks from which to sample
+ * @param sampleAmount the amount of curated artworks to insert
+ * @returns a copy of the target array with the curated artworks inserted
+ * */
+export const insertSampleCuratedWorks = (
+  target: string[],
+  curatedWorks: string[],
+  sampleAmount: number
+): string[] => {
+  const mixedArtworks = [...target]
+
+  const curatedSample = sampleSize(curatedWorks, sampleAmount)
+
+  curatedSample.forEach((challengeArtwork) => {
+    const length = target.length
+    const randomIndex = random(length)
+    mixedArtworks.splice(randomIndex, 0, challengeArtwork)
+  })
+
+  return mixedArtworks
```

Comment:
**suggestion:** This could be simplified like: 
```suggestion
  const curatedSample = sampleSize(curatedWorks, sampleAmount)
  const mixedArtworks = shuffle([...curatedWorks, ...target])

  return mixedArtworks
```

Reply:
This has come up a few times so let me know if I'm missing something. But the current goal is to preserve the order of the artworks coming back from weaviate which have been ordered by similarity. The suggestion would not preserve the order of the weaviate recommended artworks, right?

Reply:
Oh, my bad, again, I haven't been following the recent comms, and I haven't found that in the ACs in [the notion](https://www.notion.so/artsy/Infinite-Discovery-Qualitative-QA-12fcab0764a0805c921bfe0146881f9c). 
In that case you can dismiss this comment!